src/server/scripts/Spells/spell_mage.cpp

```diff
@@ -80,6 +85,106 @@ enum MiscSpells
     SPELL_MAGE_CHILLED                           = 205708
 };
 
+// 110909 - Alter Time Aura
+// 342246 - Alter Time Aura
+class spell_mage_alter_time_aura : public AuraScript
+{
+    PrepareAuraScript(spell_mage_alter_time_aura);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo
+        ({
+            SPELL_MAGE_ALTER_TIME_VISUAL,
+            SPELL_MAGE_MASTER_OF_TIME,
+            SPELL_MAGE_BLINK,
+        });
+    }
+
+    void OnApply(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        Unit* unit = GetTarget();
+        if (unit == nullptr)
+            return;
+
+        if (Player* player = unit->ToPlayer())
+        {
+            _health = player->GetHealth();
+            _loc = player->GetWorldLocation();
+        }
+    }
+
+    void AfterRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        Unit* unit = GetTarget();
+        if (unit == nullptr)
+            return;
+
+        AuraRemoveMode rmmode = GetTargetApplication()->GetRemoveMode();
+        if (rmmode == AURA_REMOVE_BY_EXPIRE)
+        {
+            if (Player* player = unit->ToPlayer())
+            {
+                if (player->GetMapId() == _loc.GetMapId())
+                {
+                    player->SetHealth(_health);
+                    player->NearTeleportTo(_loc.GetPosition());
+
+                    if (player->GetAura(SPELL_MAGE_MASTER_OF_TIME))
+                    {
+                        SpellInfo const* blink = sSpellMgr->GetSpellInfo(SPELL_MAGE_BLINK, DIFFICULTY_NONE);
+                        player->GetSpellHistory()->ResetCharges(blink->ChargeCategoryId);
+                    }
+                    player->CastSpell(player, SPELL_MAGE_ALTER_TIME_VISUAL);
+                }
+            }
+        }
+    }
+
+    void Register() override
+    {
+        OnEffectApply += AuraEffectApplyFn(spell_mage_alter_time_aura::OnApply, EFFECT_0, SPELL_AURA_OVERRIDE_ACTIONBAR_SPELLS, AURA_EFFECT_HANDLE_REAL);
+        AfterEffectRemove += AuraEffectRemoveFn(spell_mage_alter_time_aura::AfterRemove, EFFECT_0, SPELL_AURA_OVERRIDE_ACTIONBAR_SPELLS, AURA_EFFECT_HANDLE_REAL);
+    }
+
+private:
+    uint64 _health;
+    WorldLocation _loc;
+};
+
+// 127140 - Alter Time Active
+// 342247 - Alter Time Active
+class spell_mage_alter_time_active : public SpellScript
+{
+    PrepareSpellScript(spell_mage_alter_time_active);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo
+        ({
+            SPELL_MAGE_ALTER_TIME_AURA,
+            SPELL_MAGE_ARCANE_ALTER_TIME_AURA,
+        });
+    }
+
+    void RemoveAlterTimeAura(SpellEffIndex /*effIndex*/)
+    {
+        if (Unit* unit = GetCaster())
+        {
+            if (Aura* aura = unit->GetAura(SPELL_MAGE_ALTER_TIME_AURA))
+                unit->RemoveAura(aura, AURA_REMOVE_BY_EXPIRE);
```

Comment:
```suggestion
            unit->RemoveAura(SPELL_MAGE_ALTER_TIME_AURA, ObjectGuid::Empty, 0, AURA_REMOVE_BY_EXPIRE);
```

You don't have to check if aura exists before attempting to remove it